backend/src/mirrors_qa_backend/entrypoint.py

```diff
@@ -1,25 +1,62 @@
 import argparse
 import logging
+import sys
 
 from mirrors_qa_backend import logger
-from mirrors_qa_backend.db import Session
-from mirrors_qa_backend.db.mirrors import create_or_update_mirror_status
-from mirrors_qa_backend.extract import get_current_mirrors
+from mirrors_qa_backend.cli.mirrors import update_mirrors
+from mirrors_qa_backend.cli.scheduler import main as start_scheduler
+from mirrors_qa_backend.cli.worker import create_worker
 
 
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "--verbose", "-v", help="Show verbose output", action="store_true"
     )
+    subparsers = parser.add_subparsers(required=True)
+    # Sub-command for updating mirrors
+    update_mirrors_cmd = subparsers.add_parser(
+        "update-mirrors", help="Update the list of mirrors"
+    )
+    # Register the function to handle the sub-command
+    update_mirrors_cmd.set_defaults(func=update_mirrors)
 
-    args = parser.parse_args()
+    # Sub-command for starting the scheduler
+    scheduler_cmd = subparsers.add_parser(
+        "scheduler",
+        help="Start the scheduler",
+    )
+    # Register the function to handle the sub-command
+    scheduler_cmd.set_defaults(func=start_scheduler)
 
+    # Sub-command for creating a worker
+    create_worker_cmd = subparsers.add_parser(
+        "create-worker", help="Create a new worker"
+    )
+    create_worker_cmd.add_argument(
+        "worker_id", help="ID of the worker", metavar="worker-id"
+    )
+    create_worker_cmd.add_argument(
+        "--countries",
+        help="Comma-seperated country codes each in ISO 3166-1 alpha-2 format",
+        type=lambda s: s.split(","),
+    )
+    create_worker_cmd.add_argument(
+        "private_key_file",
+        metavar="private-key-file",
+        type=argparse.FileType("r", encoding="ascii"),
```

Comment:
If I used a `pathlib.Path` type, it wouldn't be able to read the contents from `stdin`. I figured one might not necessarily want to mount a file in the `backend` container before creating the worker. This would allow them to pipe input from their host to the container. Also, since I have refactored the actual function to take a `bytes` object, I can read the contents from the entrypoint and forward it to the function. `argparse.FileType` would open the file for us and I would read it in.

Reply:
Just did a little digging and the underlying ABC is an `io.Base` and I can call `read()` on the instance to read the contents until `EOF`, thus, `contents = args.private_key_file.read()`

Reply:
That works 👍